lib/parsers/design_token_parser.dart

```diff
@@ -0,0 +1,115 @@
+import 'package:design_tokens_builder/builder_config/builder_config.dart';
+import 'package:design_tokens_builder/parsers/border_parser.dart';
+import 'package:design_tokens_builder/parsers/border_radius_parser.dart';
+import 'package:design_tokens_builder/parsers/box_shadow_parser.dart';
+import 'package:design_tokens_builder/parsers/color_parser.dart';
+import 'package:design_tokens_builder/parsers/dimension_parser.dart';
+import 'package:design_tokens_builder/parsers/font_family_parser.dart';
+import 'package:design_tokens_builder/parsers/font_weight_parser.dart';
+import 'package:design_tokens_builder/parsers/line_height_parser.dart';
+import 'package:design_tokens_builder/parsers/number_parser.dart';
+import 'package:design_tokens_builder/parsers/opacity_parser.dart';
+import 'package:design_tokens_builder/parsers/spacing_parser.dart';
+import 'package:design_tokens_builder/parsers/text_case_parser.dart';
+import 'package:design_tokens_builder/parsers/text_decoration_parser.dart';
+import 'package:design_tokens_builder/parsers/typography_parser.dart';
+import 'package:design_tokens_builder/utils/string_utils.dart';
+
+/// Returns the [DesignTokenParser] for the given type.
+///
+/// Therefore the method checks [DesignTokenParser.tokenType] array if it
+/// contains the passed `type`.
+/// Make to register one [DesignTokenParser] per type. Otherwise the wrong
+/// parser might be returned.
+DesignTokenParser parserForType(
+  String type, {
+  int indentationLevel = 1,
+  BuilderConfig? config,
+}) {
+  try {
+    return [
+      BorderParser(indentationLevel, config),
+      BorderRadiusParser(indentationLevel, config),
+      BoxShadowParser(indentationLevel, config),
+      ColorParser(indentationLevel, config),
+      DimensionParser(indentationLevel, config),
+      FontFamilyParser(indentationLevel, config),
+      FontWeightParser(indentationLevel, config),
+      LineHeightParser(indentationLevel, config),
+      NumberParser(indentationLevel, config),
+      OpacityParser(indentationLevel, config),
+      SpacingParser(indentationLevel, config),
+      TextCaseParser(indentationLevel, config),
+      TextDecorationParser(indentationLevel, config),
+      TypographyParser(indentationLevel, config),
+    ].firstWhere((element) => element.tokenType.contains(type));
```

Comment:
This solution requires building whole list of parsers every time the token is going to be parsed. That may create a bit of overhead.

I'd suggest to make a DesignTokenFactory that keeps a mapping between tokenType and the DesignTokenParser constructor. That way this list could be created once and factory would create an expected `DesignTokenParser` for a given type. The problem is that the factory would have to be updated whenever new parser is added.

Reply:
Is that even possible? Because the indentationLevel can be different each call 🤔 So we would need to build the list nevertheless every time we get a parser, no?

Reply:
You need to separate mapping from type to parser from actual parse method. Mapping logic goes to factory instance so indentation would be parameter of `parse`.